app/components/money-display.tsx

```diff
@@ -0,0 +1,153 @@
+import type { ReactNode } from 'react';
+import type { Currency, CurrencyUnit, Money } from '~/lib/money';
+import { cn } from '~/lib/utils';
+
+type MoneyDisplayBaseProps = {
+  /** The symbol to display */
+  symbol: string;
+  /** Place the symbol before or after the number */
+  symbolPosition: 'prefix' | 'suffix';
+  /** The number to display */
+  children: ReactNode;
+  size?: 'sm' | 'default';
+  className?: string;
+};
+
+const sizes = {
+  sm: {
+    symbol: 'text-[1.2rem]',
+    value: 'text-2xl pt-1',
+  },
+  default: {
+    symbol: 'text-[3.45rem]',
+    value: 'text-6xl pt-2',
+  },
+} as const;
+
+function MoneyDisplayBase({
+  symbol,
+  symbolPosition,
+  size = 'default',
+  className,
+  children,
+}: MoneyDisplayBaseProps) {
+  return (
+    <div className={cn('inline-flex w-fit items-center', className)}>
+      {symbolPosition === 'prefix' && (
+        <span className={cn('font-bold', sizes[size].symbol)}>{symbol}</span>
+      )}
+      <span className={cn('font-bold font-numeric', sizes[size].value)}>
+        {children}
+      </span>
+      {symbolPosition === 'suffix' && (
+        <span className={cn('font-bold', sizes[size].symbol)}>{symbol}</span>
+      )}
+    </div>
+  );
+}
+
+type MoneyDisplayProps<C extends Currency> = Omit<
+  MoneyDisplayBaseProps,
+  'symbol' | 'symbolPosition' | 'children'
+> & {
+  /** The money to display */
+  money: Money<C>;
+  /**
+   * The number of decimal places entered
+   *
+   * If undefined, then no decimal places are shown.
+   *
+   * If not undefined, then we want to grey out the remaining decimal places.
+   *
+   * The difference between this value and the number of decimal places in the money object
+   * is used to grey out the zeros.
+   */
+  decimalPlaces?: number;
+  /**
+   * If true, then `decimalPlaces` is ignored and the number is displayed as money.toLocaleString()
+   */
+  simpleFormat?: boolean;
```

Comment:
this makes it so that we can use this component for static amounts and input amounts